tests/test_cli.py

```diff
@@ -517,3 +529,18 @@ def test_run_non_universal_newline(self):
             '\n' % path)
         self.assertEqual(
             (ctx.returncode, ctx.stdout, ctx.stderr), (1, expected_out, ''))
+
+    def test_encoding_detection_utf16le(self):
+        path = os.path.join(self.wd, 'non-ascii/utf16le')
+        encoding = cli.determine_encoding(path)
+        self.assertEqual(encoding, 'utf-16-le')
```

Comment:
Can you also run `yamllint` on these files, just to make sure everything works fine?
```suggestion
        self.assertEqual(encoding, 'utf-16-le')

        with RunContext(self) as ctx:
            cli.run(())
        self.assertNotEqual(ctx.returncode, 0)
        self.assertEqual(ctx.stdout, '')
        self.assertEqual(ctx.stderr, '')
```
(same for other encodings)